mirrord-agent/src/outgoing.rs

```diff
@@ -42,6 +43,84 @@ pub(crate) struct TcpOutgoingApi {
     daemon_rx: Receiver<Daemon>,
 }
 
+#[tracing::instrument(skip(allocator, writers, readers, daemon_tx))]
+async fn layer_recv(
+    layer_message: LayerTcpOutgoing,
+    allocator: &mut IndexAllocator<ConnectionId>,
+    writers: &mut HashMap<ConnectionId, OwnedWriteHalf>,
+    readers: &mut StreamMap<ConnectionId, ReaderStream<OwnedReadHalf>>,
+    daemon_tx: Sender<DaemonTcpOutgoing>,
+) -> Result<(), AgentError> {
+    match layer_message {
+        // [user] -> [layer] -> [agent] -> [layer]
+        // `user` is asking us to connect to some remote host.
+        LayerTcpOutgoing::Connect(LayerConnect { remote_address }) => {
+            let daemon_connect = timeout(
+                Duration::from_millis(1500),
```

Comment:
How'd you come up with 1500? Seems like it might be too short?

Reply:
Increased to 3 seconds. Should I create a more robust mechanism for increasing this timeout?

Reply:
I'm not sure - what are the repercussions here? For example, if the user's application intentionally makes an outgoing request that is long-running, are we going to fail it?

Reply:
Yeah, it'll fail.

So the question is, should I address this now, or wait until someone opens up an issue?

Reply:
I'm leaning towards option B here, but do you have a solution in mind?

Reply:
Hmm I could investigate more deeply what's blocking the `connect` call for this specific case in go (while in all other cases it behaves as expected).

Reply:
I wouldn't worry about it for now, but would add some documentation + a TODO

Reply:
TODO added.